test/unit/cuda/test-reduce-max.cpp

```diff
@@ -0,0 +1,221 @@
+//
+// Produced at the Lawrence Livermore National Laboratory
+//
+// LLNL-CODE-689114
+//
+// All rights reserved.
+//
+// This file is part of RAJA.
+//
+// For additional details, please also read RAJA/README.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice,
+//   this list of conditions and the disclaimer below.
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the disclaimer (as noted below) in the
+//   documentation and/or other materials provided with the distribution.
+//
+// * Neither the name of the LLNS/LLNL nor the names of its contributors may
+//   be used to endorse or promote products derived from this software without
+//   specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL LAWRENCE LIVERMORE NATIONAL SECURITY,
+// LLC, THE U.S. DEPARTMENT OF ENERGY OR CONTRIBUTORS BE LIABLE FOR ANY
+// DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+// OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+// HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+// IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+///
+/// Source file containing tests for RAJA GPU max reductions.
+///
+
+#include <cfloat>
+#include <cstdio>
+#include <iostream>
+#include <random>
+#include <string>
+
+#include "RAJA/RAJA.hpp"
+#include "gtest/gtest.h"
+
+#define TEST_VEC_LEN 1024 * 1024 * 6
+
+using namespace RAJA;
+
+const size_t block_size = 256;
+
+const int test_repeat = 10;
+
+std::random_device rd;
+std::mt19937 mt(rd());
+std::uniform_real_distribution<double> dist(-10, 10);
+std::uniform_real_distribution<double> dist2(0, TEST_VEC_LEN - 1);
+
+double *getData()
+{
+  double *dvalue;
+  cudaMallocManaged((void **)&dvalue,
+                    sizeof(double) * TEST_VEC_LEN,
+                    cudaMemAttachGlobal);
+  for (int i = 0; i < TEST_VEC_LEN; ++i) {
+    dvalue[i] = -DBL_MAX;
+  }
+  return dvalue;
+}
+
+//
+// test 1 runs 3 reductions over a range multiple times to check
+//        that reduction value can be retrieved and then subsequent
+//        reductions can be run with the same reduction objects.
+//        Also exercises the get function call
+void test1()
```

Comment:
I didn't really **want** to do it this way, but gtest does the weird class scoping which breaks NVCC.